public/js/models.js

```diff
@@ -88,11 +88,20 @@ function computeInference(faceKeyPoints, handKeyPoints) {
     return Promise.reject(Error('Both key points must be set'));
   }
 
-  const fp = tf.tensor(faceKeyPoints).expandDims(0);
-  const hp = tf.tensor(handKeyPoints).expandDims(0);
+  let fp = tf.tensor(faceKeyPoints).expandDims(0);
+  let hp = tf.tensor(handKeyPoints).expandDims(0);
+
+  // Compute difference between every (hp, fp) pair
+  hp = hp.expandDims(2);
+  fp = fp.expandDims(1);
```

Comment:
Nit: is there a reason why you're essentially redoing the `expandDims` calculations?
Wouldn't something like this be more efficient and still allow the variables to be immutable:
```js
const fp = tf
  .tensor(faceKeyPoints)
  .expandDims(0)
  .expandDims(1);
// or const fp = tf.tensor(faceKeyPoints).expandDims(0).expandDims(1);
```

Reply:
great question; 2 reasons:
1) it's a bit more readable/understandable if we do the 2 logical operations (1. adding batch dimension and 2. broadcasting to compute the pair-wise differences) separately - it shouldn't have a large effect on performance since doing dimension manipulation doesn't trigger a memcopy in the underlying runtime.
1) eventually lines 94-101 will need to be factored out (as hinted to by the note at top of the PR) but lines 91-92 will stay.

Reply:
Oh okay, I see, thanks for the clarifications.